pwem/viewers/viewer_chimera.py

```diff
@@ -316,61 +317,63 @@ def show(self):
+
                 f = open(fnCmd, 'w')
+                f.write('cd %s\n' % os.getcwd())
                 f.write("cofr 0,0,0\n")  # set center of coordinates
-                if obj.hasVolume():
+                f.write("style stick\n")
+
+                _inputVol = obj.getFirstItem().getVolume()
+                if _inputVol is not None:
                     volID = 1
-                    volumeObject = obj.getVolume()
-                    dim = volumeObject.getDim()[0]
-                    sampling = volumeObject.getSamplingRate()
-                    f.write("open %s\n" % os.path.abspath(
-                        emlib.image.ImageHandler.removeFileType(volumeObject.getFileName())))
-                    f.write("volume #%d style surface voxelSize %f\n"
-                            % (volID, sampling))
-                    x, y, z = volumeObject.getShiftsFromOrigin()
-                    f.write("volume #%d origin %0.2f,%0.2f,%0.2f\n"
-                            % (volID, x, y, z))
+                    dim, sampling = _inputVol.getDim()[0], _inputVol.getSamplingRate()
+
+                    f.write("open %s\n" % _inputVol.getFileName())
+                    x, y, z = _inputVol.getOrigin(force=True).getShifts()
+                    f.write("volume #%d style surface voxelSize %f\nvolume #%d origin "
+                            "%0.2f,%0.2f,%0.2f\n"
+                            % (volID, sampling, volID, x, y, z))
                 else:
-                    dim = 150  # eventually we will create a PDB library that
-                    # computes PDB dim
-                    sampling = 1.
-                # Construct the coordinate file
-                bildFileName = os.path.abspath(
-                    os.path.join(extraPath, "axis.bild"))
-                Chimera.createCoordinateAxisFile(dim,
-                                                 bildFileName=bildFileName,
-                                                 sampling=sampling)
+                    dim, sampling = 150., 1.
+
+                bildFileName = self.protocol._getExtraPath("axis_output.bild")
+                Chimera.createCoordinateAxisFile(dim, bildFileName=bildFileName, sampling=sampling)
                 f.write("open %s\n" % bildFileName)
-                f.write("open %s\n" % os.path.abspath(fn))
-                f.write("style stick\n")
-                f.write("view\n")
+
+                for AS in obj:
+                    f.write("open %s\n" % AS.getFileName())
+                    #f.write("style stick\n")
+
                 f.close()
-                ChimeraView(fnCmd).show()
-            # FIXME: there is an asymmetry between ProtocolViewer and Viewer
-            # for the first, the visualize method return a list of View's
-            # (that are shown)
-            # for the second, the visualize directly shows the objects.
-            # the first approach is better
+                # run in the background
```

Comment:
Visualize method should return a view not running anything. Although it may work is not the right way to do it. 

Not important, but in the past, when the webtools were available, and in case in the "distant" future we provide a web interface, this runChimeraProgram is a problem.